core/domain/services/admin/events/data/PreviewDeletion.php

```diff
@@ -0,0 +1,141 @@
+<?php
+
+namespace EventEspresso\core\domain\services\admin\events\data;
+
+use EE_Admin_Page;
+use EEH_Template;
+use EEM_Datetime;
+use EEM_Event;
+use EEM_Registration;
+use EventEspresso\admin_pages\events\form_sections\ConfirmEventDeletionForm;
+use EventEspresso\core\exceptions\UnexpectedEntityException;
+use EventEspresso\core\services\orm\tree_traversal\NodeGroupDao;
+use Events_Admin_Page;
+
+/**
+ * Class PreviewDeletion
+ *
+ * Description
+ *
+ * @package     Event Espresso
+ * @author         Mike Nelson
+ * @since         $VID:$
+ *
+ */
+class PreviewDeletion
+{
+    /**
+     * @var NodeGroupDao
+     */
+    protected $dao;
+
+    /**
+     * @var EEM_Event
+     */
+    protected $event_model;
+
+    /**
+     * @var EEM_Datetime
+     */
+    protected $datetime_model;
+
+    /**
+     * @var EEM_Registration
+     */
+    protected $registration_model;
+
+    /**
+     * PreviewDeletion constructor.
+     * @param NodeGroupDao $dao
+     * @param EEM_Event $event_model
+     * @param EEM_Datetime $datetime_model
+     * @param EEM_Registration $registration_model
+     */
+    public function __construct(
+        NodeGroupDao $dao,
+        EEM_Event $event_model,
+        EEM_Datetime $datetime_model,
+        EEM_Registration $registration_model
+    )
+    {
+        $this->dao = $dao;
+        $this->event_model = $event_model;
+        $this->datetime_model = $datetime_model;
+        $this->registration_model = $registration_model;
+    }
+
+    public function handle(Events_Admin_Page $admin_page)
```

Comment:
I'm not injecting the admin page because it ended up instantiating a SECOND admin page, which caused warnings from the constants already being defined the first time. I avoided that can of worms by just passing it as a variable to this method.

Reply:
ya I would have just passed the `$request_data` array and the URL returned from `$admin_page->admin_base_url()` as parameters and then have this class return an array of template args and let `Events_Admin_Page` handle setting them and loading the actual view.

ie something like:

```php
    protected function previewDeletion()
    {
        $preview_deletion = $this->getLoader()->getShared(
			'\EventEspresso\core\domain\services\admin\events\data\PreviewDeletion'
		);
        $template_args = $preview_deletion->handle(
			$this>get_request_data(),
			$this->admin_base_url()
		);
		$this->set_template_args( $template_args );
		$this->display_admin_page_with_no_sidebar();
    }
```

But this works as is and I don't want to hold it up any more